aws-athena/views/reporting-vw_ratio_stats.sql

```diff
@@ -60,84 +22,54 @@ model_values AS (
-            WHEN asmt_all.procname = 'BORVALUE' THEN 'bor certified'
-        END AS assessment_stage,
-        MAX(
-            CASE
-                WHEN asmt_all.taxyr < '2020' THEN asmt_all.ovrvalasm3
-                WHEN asmt_all.taxyr >= '2020' THEN asmt_all.valasm3
-            END
-        ) * 10 AS total
-    FROM {{ source('iasworld', 'asmt_all') }} AS asmt_all
-    WHERE (asmt_all.valclass IS NULL OR asmt_all.taxyr < '2020')
-        AND asmt_all.procname IN ('CCAOVALUE', 'CCAOFINAL', 'BORVALUE')
-        AND asmt_all.taxyr >= '2021'
-        AND asmt_all.deactivat IS NULL
-    GROUP BY
-        asmt_all.parid,
-        asmt_all.taxyr,
-        asmt_all.procname,
-        CASE
-            WHEN asmt_all.procname = 'CCAOVALUE' THEN 'mailed'
-            WHEN asmt_all.procname = 'CCAOFINAL' THEN 'assessor certified'
-            WHEN asmt_all.procname = 'BORVALUE' THEN 'bor certified'
-        END
+        pin,
+        year,
+        class,
+        LOWER(stage_name) AS assessment_stage,
+        tot * 10 AS total
+    FROM {{ ref('reporting.vw_pin_value_long') }}
+    WHERE year >= '2021'
 ),
 
 all_values AS (
     SELECT * FROM model_values
     UNION
     SELECT * FROM iasworld_values
-),
-
-parking_space AS (
-    SELECT *
-    FROM {{ ref('default.vw_pin_condo_char') }}
-    WHERE is_parking_space = TRUE
 )
 
 SELECT
     vwps.pin,
     av.year,
     vwps.year AS sale_year,
-    classes.property_group,
+    towns.property_group,
     av.assessment_stage,
-    town_names.triad,
-    townships.township_code,
+    towns.triad_code AS triad,
+    towns.township_code,
     av.total AS fmv,
     vwps.sale_price,
     av.total / vwps.sale_price AS ratio
 FROM {{ ref('default.vw_pin_sale') }} AS vwps
-LEFT JOIN classes
-    ON vwps.pin = classes.parid
-    AND vwps.year = classes.taxyr
-LEFT JOIN townships
-    ON vwps.pin = townships.parid
-    AND vwps.year = townships.taxyr
-LEFT JOIN town_names
-    ON townships.township_code = town_names.township_code
+LEFT JOIN {{ ref('reporting.vw_pin_township_class') }} AS towns
+    ON vwps.pin = towns.pin
+    AND vwps.year = towns.year
     -- Join sales so that values for a given year can be compared to a
     -- complete set of sales from the previous year
 INNER JOIN all_values AS av
-    ON vwps.pin = av.parid
+    ON vwps.pin = av.pin
     AND CAST(vwps.year AS INT) = CAST(av.year AS INT) - 1
     -- Grab parking spaces and join them to aggregate stats for removal
-LEFT JOIN parking_space AS ps
-    ON av.parid = ps.pin
+LEFT JOIN {{ ref('default.vw_pin_condo_char') }} AS ps
+    ON av.pin = ps.pin
     AND av.year = ps.year
+    AND ps.is_parking_space = TRUE
 WHERE NOT vwps.is_multisale
     AND NOT vwps.sale_filter_deed_type
     AND NOT vwps.sale_filter_less_than_10k
     AND NOT vwps.sale_filter_same_sale_within_365
-    AND classes.property_group IS NOT NULL
+    AND towns.property_group IS NOT NULL
     AND ps.pin IS NULL
```

Comment:
**question:** This is just an anti-join to remove parking spaces right?

Reply:
Yep.